aten/src/ATen/Context.cpp

```diff
@@ -142,9 +143,47 @@ bool Context::allowTF32CuBLAS() const {
 }
 
 void Context::setAllowTF32CuBLAS(bool b) {
+  TORCH_WARN("Setting allow_tf32_cublas directly is not recommend as it may conflict with"
```

Comment:
I don't think we need to warn here, people can set either and it's fine

Reply:
@mruberry our concern was to allow for conflicting "configs", e.g. setting the `Float32MatmulPrecision` to `HIGHEST`, while changing the `allow_tf32` flag to `True`. In this case the user is creating a "custom" precision level and depending which value the user is reading a conflicting behavior could be seen (i.e. checking the matmul precision vs. the `allow_tf32` flag).

Reply:
`allow_tf32` getter and matmul precision getter should read the same underlying variable. Same for setters. So either function can be used, whichever was called later sets the precision.

Reply:
I like this suggestion but I'm not sure what the behavior should be e.g., if the precision was set to `HIGHEST` and then `allow_tf32` is set to `True`. Would we do `HIGH`, or `MEDIUM`?

Reply:
Talked to @ngimel offline and we agree with you, @eqy. We still don't want to bother warning here. 

Let's just have the getter return the last set value (the default value if nothing has ever been set). Ops need to read the specific flags, and in the future when there are multiple flags users might configure the general precision desired using this flag, then customize for particular hardware as needed, and that's OK. Today it's true that if a user sets the `allow_tf32` flag directly and then gets the value of the generic flag then they are "out of sync" in some sense, but it seems very unlikely that programs will do this.

Reply:
Sure, that sounds reasonable. Rather than warning each time, would it make sense to warn if the setting would take things "out of sync?"

Reply:
We should have one controlling variable, then nothing could possibly go out of sync ever (except people calling different setting on the different threads, but that would lead to unpredictable results even today, so there's no regression). And we won't need warning.

Reply:
With one variable (if we only kept track of `allow_tf32`), setting `MEDIUM` or `HIGH` matmul precision is doable as they have the same observable effect, but would it be strange if they both mapped to one setting when the getter was called?

If we only kept track of `float32_matmul_precision`, we have the opposite decision to make of whether `allow_tf32=false` should map to `HIGH` or `MEDIUM`.

Reply:
Yeah unfortunately allow_tf32 won't be enough because it has 2 values instead of 3. For `float32_matmul_precision` we can map it to `HIGH`.